### PR TITLE
Update group types for DsyncGroupUser events

### PR DESCRIPTION
**Pull request by awolfden**
## Description
Update group types to return full group object for DsyncGroupUser events

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: src/common/interfaces/event.interface.ts [hunk 1]
```diff
@@ -113,7 +113,7 @@ export interface DsyncGroupUserAddedEvent extends EventBase {
   data: {
     directoryId: string;
     user: DirectoryUser;
-    group: Pick<EventDirectoryGroup, 'id' | 'name'>;
+    group: EventDirectoryGroupResponse;
```
Comment (willmanduffy) [inline src/common/interfaces/event.interface.ts:116]: This should be an `EventDirectoryGroup`

File: src/common/interfaces/event.interface.ts [hunk 2]
```diff
@@ -131,7 +131,7 @@ export interface DsyncGroupUserRemovedEvent extends EventBase {
...
```
Comment (willmanduffy) [inline src/common/interfaces/event.interface.ts:134]: This should be an `EventDirectoryGroup`

File: src/common/interfaces/event.interface.ts (hunk 1 above)
Comment (awolfden) [reply src/common/interfaces/event.interface.ts]: Yep, good call - initially these were both throwing errors with EventDirectoryGroupResponse, but I just needed to adjust the event.serializer.ts file a bit to make it work.

File: src/common/serializers/event.serializer.ts [hunk 3]
```diff
@@ -58,7 +58,7 @@ export const deserializeEvent = (event: EventResponse): Event => {
         data: {
           directoryId: event.data.directory_id,
           user: deserializeDirectoryUser(event.data.user),
-          group: event.data.group,
+          group: deserializeEventDirectoryGroup(event.data.group),
```
Comment (willmanduffy) [inline src/common/serializers/event.serializer.ts:61]: Actually looks like `EventDirectoryGroup` and `DirectoryGroup` are now identical. Could we make that full migration and also make this `deserializeDirectoryGroup`?

Sorry missed this before

File: package-lock.json [hunk 4]
```diff
@@ -0,0 +1,3167 @@
+{
```
Comment (willmanduffy) [inline package-lock.json:1]: We won't want to commit this

Comment (awolfden) [reply package-lock.json:1]: 🤦🏽

File: src/common/serializers/event.serializer.ts (hunk 3 above)
Comment (awolfden) [reply src/common/serializers/event.serializer.ts]: ✅

File: src/common/interfaces/event.interface.ts [hunk 5]
```diff
@@ -122,7 +122,7 @@ export interface DsyncGroupUserAddedEventResponse extends EventResponseBase {
   data: {
     directory_id: string;
     user: DirectoryUserResponse;
-    group: Pick<EventDirectoryGroupResponse, 'id' | 'name'>;
+    group: EventDirectoryGroupResponse;
```
Comment (willmanduffy) [inline src/common/interfaces/event.interface.ts:125]: Is `EventDirectoryGroupResponse` the same as `DirectoryGroupResponse` at this point?

Comment (awolfden) [reply src/common/interfaces/event.interface.ts]: Sure is! Removed the event version in favor of DirectoryGroupResponse.

File: src/common/serializers/event.serializer.ts [hunk 6]
```diff
@@ -58,7 +58,7 @@ export const deserializeEvent = (event: EventResponse): Event => {
         data: {
           directoryId: event.data.directory_id,
           user: deserializeDirectoryUser(event.data.user),
-          group: event.data.group,
+          group: deserializeDirectoryGroup(event.data.group),
```
[the remark this answers is not in the document]
Comment (awolfden) [reply src/common/serializers/event.serializer.ts:61]: We can and we will 💪🏽

**Review by willmanduffy — APPROVED** [review]
Looks like there's a formatting problem

**Pull request merged**